### ISSUE TITLE
Issue with webpack live-reload

### ISSUE THREAD

**Issue by PavloGlazunov** [Typescript-TDD/ts-auto-mock#105]
I did and run a simple example usage of this library. I faced with runtime issue after I re-save my code then I get an error:
> TypeError: c_projects_test_src_index_ts_Repository.ɵRepository.instance.getFactory(...) is not a function

**Comment by uittorio** [comment]
Hi @PavloGlazunov, thanks for reporting this issue. 

In order to investigate it will be easier for us to see your example and how its configured. Do you have a repo on github that we can double check?

**Comment by PavloGlazunov** [comment]
> Hi @PavloGlazunov, thanks for reporting this issue.
> 
> In order to investigate it will be easier for us to see your example and how its configured. Do you have a repo on github that we can double check?

Yes, I have. Please, look [here](https://github.com/PavloGlazunov/TestTypescriptMocking)

**Comment by Pmyl** [comment]
Thanks for the repo @PavloGlazunov !

I see that you’re using a standard webpack configuration without any test runner, live reload in that case is something we never really tested.

As far as I remember using karma watch the single file recompilation works fine so I hope this won’t be too difficult to make it work.

I’ll have a look as soon as possible

**Comment by PavloGlazunov** [comment]
@Pmyl Thank you for your response. Please, keep me informed.

**Comment by Pmyl** [comment]
@PavloGlazunov we have mixed news unfortunately...

The main issue here is that our internal caching doesn't work well with repeated compilation, we already have a solution in mind and we will try to implement it as soon as possible.
In the meantime the solution is to disable the cache using the configuration property `cacheBetweenTests` in this way:

```ts
options: {
  getCustomTransformers: function (program) {
    return ({
      before: [
        tsAutoMockTransformer(program, { cacheBetweenTests: false })
      ]
    });
  }
}
```

There is another issue here though... `ts-loader` is not acting nice with transformers in live reload mode, if you change a contract file it doesn't update the AST for the test file causing the test file to have the old mock instead of the updated one (for example if you add a property on the interface).

I tried it with the most basic transformer (our nice inspiration ts-transformers-keys) and it act in the same way unfortunately so it's not our library's fault.
I would recommend to use `awesome-typescript-loader` to run it, the configuration is identical and everything works as intended.
